sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/sources/ForeachWriterSuite.scala

```diff
@@ -142,8 +142,12 @@ class ForeachWriterSuite extends StreamTest with SharedSparkSession with BeforeA
       val e = intercept[StreamingQueryException] {
         query.processAllAvailable()
       }
-      assert(e.getCause.isInstanceOf[SparkException])
-      assert(e.getCause.getCause.getMessage === "ForeachSinkSuite error")
+
+      val errClass = "FOREACH_USER_FUNCTION_ERROR"
+
+      // verify that we classified the exception
+      assert(e.getMessage.contains(errClass))
```

Comment:
you can use `checkError()` func

Reply:
Thanks for your review! I am afraid we cannot use checkError here as the errorClass that e returns is `STREAM_FAILED` and the `FOREACH_USER_FUNCTION_ERROR` was thrown as part of the message instead of cause [here](https://github.com/apache/spark/pull/45449/files#diff-7e8a99d76adb60a2d85609ab46bb2867cd83de106b3c048d6d955805824c9bb6R335) .

Reply:
But I guess we should be still able to figure out error class from cause, right? Otherwise how we could leverage the new error class we added in this fix?

e.cause should be wrapped exception having `FOREACH_USER_FUNCTION_ERROR` error class. Otherwise we'll need to revisit whether the error class will be captured in termination event in streaming query listener.

Reply:
Now I just remembered. We didn't add more element into exception stack when dealing with ForeachBatchUserFuncException, and you're following this. Got it.

Reply:
Yeah, it is following the style that Micheal did in this PR: https://github.com/apache/spark/pull/45299